tests/scenarios/v2-addon-dev-watch-test.ts

```diff
@@ -134,6 +137,139 @@ Scenarios.fromProject(() => baseV2Addon())
+          let demoContent = await fs.readFile(demoHbs);
+
+          // deleting a component from src should delete it from dist
+          await fs.rm(srcPathButton);
+          await watcher?.nextBuild();
+          assert.strictEqual(existsSync(distPathButton), false, `Expected ${distPathButton} to be deleted`);
+
+          // create a component in src should create it in dist
+          await fs.writeFile(srcPathButton, origContent);
+          await watcher?.nextBuild();
+          assert.strictEqual(existsSync(distPathButton), true, `Expected ${distPathButton} to exist`);
+
+          // updating hbs modifies colocated js
+          await becomesModified({
+            filePath: distPathDemoComp,
+            assert,
+            // Update a component
+            fn: async () => {
+              let someContent = await fs.readFile(demoHbs);
+
+              // generally it's bad to introduce time dependencies to a test, but we need to wait long enough
+              // to guess for how long it'll take for the file system to update our file.
+              //
+              // the `stat` is measured in `ms`, so it's still pretty fast
+              await fs.writeFile(demoHbs, someContent + `\n`);
+              await aBit(10);
+              await watcher?.nextBuild();
+            },
+          });
+
+          // removing hbs modifies colocated js to not import hbs anymore
+          await becomesModified({
+            filePath: distPathDemoComp,
+            assert,
+            // Update a component
+            fn: async () => {
+              // generally it's bad to introduce time dependencies to a test, but we need to wait long enough
+              // to guess for how long it'll take for the file system to update our file.
+              //
+              // the `stat` is measured in `ms`, so it's still pretty fast
+              await fs.rm(demoHbs);
+              await aBit(10);
+              await watcher?.nextBuild();
+            },
+          });
+
+          await fs.writeFile(demoHbs, demoContent);
+          await watcher?.nextBuild();
+
+          // updating hbs content should not update unrelated files
+          await isNotModified({
+            filePath: distPath,
+            assert,
+            // Update a component
+            fn: async () => {
+              let someContent = await fs.readFile(demoHbs);
+
+              // generally it's bad to introduce time dependencies to a test, but we need to wait long enough
+              // to guess for how long it'll take for the file system to update our file.
+              //
+              // the `stat` is measured in `ms`, so it's still pretty fast
+              await fs.writeFile(demoHbs, someContent + `\n\n`);
+              await aBit(10);
+              await watcher?.nextBuild();
+            },
+          });
+
+          // updating hbs content should update resulting app re-exported component
```

Comment:
Is this a typo, or am I misunderstanding?

```suggestion
          // updating hbs content should not update resulting app re-exported component
```